lib/rubocop/cop/rails/env.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Rails
+      # This cop checks typo of environment name.
+      #
+      # @example
+      #   # bad
+      #   Rails.env.proudction?
+      #
+      #   # good
+      #   Rails.env.production?
+      class Env < Cop
+        include NameSimilarity
+
+        MSG = '`%<name>s` is incorrect environment name.'.freeze
```

Comment:
Would probably say "unknown" instead of "incorrect", as it gives a better hint what's wrong. 🙂 

E.g.: "Unknown environment `prduction`. Did you mean `production`?"